app/models/note.rb

```diff
@@ -0,0 +1,8 @@
+class Note < ActiveRecord::Base
+  attr_accessible :title, :content, :entity_id, :entity_type
+
+  belongs_to :entity, polymorphic: true
```

Comment:
when doing a polymorphic belongs_to, I always find it useful to validate the `entity_type` so by looking at the note model you can tell which models can be a entity.

```suggestion
  belongs_to :entity, polymorphic: true
  validates :entity_type, inclusion: ['Student', 'Course', 'Institute']
```